meteor.js

```diff
@@ -1,244 +1,318 @@
-            })
-            css = css.join("\n        ");
-            content = content.replace(/{{ *> *css *}}/, css);
-
-            // ADD the SCRIPT files
-            var scripts = ['__meteor_runtime_config__'];
-            _.each(files['js'], function(jsFile) {
-                scripts.push('<script type="text/javascript" src="'+ jsFile +'"></script>');
-            })
-            scripts = scripts.join("\n        ");
-
-            // add the meteor runtime config
-            settings = {
-                'meteorRelease': starJson.meteorRelease,
-                'ROOT_URL_PATH_PREFIX': '',
-                meteorEnv: { NODE_ENV: 'production' },
-                autoupdate: { versions : {}},
-                // 'DDP_DEFAULT_CONNECTION_URL': program.url || '', // will reload infinite if Meteor.disconnect is not called
-                // 'appId': process.env.APP_ID || null,
-                // 'autoupdateVersion': null, // "ecf7fcc2e3d4696ea099fdd287dfa56068a692ec"
-                // 'autoupdateVersionRefreshable': null, // "c5600e68d4f2f5b920340f777e3bfc4297127d6e"
-                // 'autoupdateVersionCordova': null
-            };
-            // on url = "default", we dont set the ROOT_URL, so Meteor chooses the app serving url for its DDP connection
-            if(program.url !== 'default')
-                settings.ROOT_URL = program.url || '';
-
-
-            if(settingsJson.public)
-                settings.PUBLIC_SETTINGS = settingsJson.public;
-
-            scripts = scripts.replace('__meteor_runtime_config__', '<script type="text/javascript">__meteor_runtime_config__ = JSON.parse(decodeURIComponent("'+encodeURIComponent(JSON.stringify(settings))+'"));</script>');
-            
-            // add Meteor.disconnect() when no server is given
-            if(!program.url)
-                scripts += "\n"+'        <script type="text/javascript">Meteor.disconnect();</script>';
-
-            content = content.replace(/{{ *> *scripts *}}/, scripts);
-
-            // write the index.html
-            return fs.writeFileAsync(path.join(buildPath, 'index.html'), content);
-        });
-    },
-    cleanUp: function(callback) {
-        return Q.try(function() {
-            // remove files
-            deleteFolderRecursive(path.join(buildPath, 'bundle'));
-            fs.unlinkSync(path.join(buildPath, 'program.json'));
-            try{
-                fs.unlinkSync(path.join(buildPath, 'head.html'));
-            } catch (e){
-                console.log("Didn't unlink head.html; doesn't exist.");
-            }            
+      // ADD HEAD
+      content = content.replace(/{{ *> *head *}}/, head);
+      content = content.replace(/{{ *> *body *}}/, body);
+
+      // get the css and js files
+      const files = {
+        css: [],
+        js: []
+      };
+
+      _.each(fs.readdirSync(outputPath), (file) => {
+        if (/^[a-z0-9]{40}\.js$/.test(file)) {
```

Comment:
> Maybe putting the regex into a const in higher level scope?

👍👍👍


> Will the file definitely without any special characters, like -? I suppose this will be the case for generated js scripts?

So far it's hash-string, no special symbols.

Reply:
> Maybe putting the regex into a const in higher level scope?

Done here: https://github.com/frozeman/meteor-build-client/commit/1b083cda76d4746fc6d55923861acc843d9bf5b9